Scripts/RemediationScripts/Remediate-SetSQLServerMinReqTLSVersion.ps1

```diff
@@ -0,0 +1,999 @@
+        Write-Host $([Constants]::SingleDashLine)
+        $controlForRemediation = Get-content -path $Path | ConvertFrom-Json
+        $controls = $controlForRemediation.ControlRemediationList
+        $resourceDetails = $controls | Where-Object { $controlIds -eq $_.ControlId };
+        $validResources = $resourceDetails.FailedResourceList | Where-Object {![String]::IsNullOrWhiteSpace($_.ResourceId)}
+
+        if(($resourceDetails | Measure-Object).Count -eq 0 -or ($validResources | Measure-Object).Count -eq 0)
+        {
+            Write-Host "No SQL Server(s) found in input json file for remediation." -ForegroundColor $([Constants]::MessageType.Error)
+            Write-Host $([Constants]::DoubleDashLine)
+            return
+        } 
+
+        $validResources | ForEach-Object { 
+            try
+            {
+                $name = $_.ResourceName
+                $resourceGroupName = $_.ResourceGroupName
+                $resSqlServer = Get-AzSqlServer  -Name $name -ResourceGroupName $resourceGroupName  -ErrorAction SilentlyContinue
+                $sqlServerResources = $sqlServerResources + $resSqlServer
+            }
+            catch
+            {
+                Write-Host "Valid resource id(s) not found in input json file. Error: [$($_)]" -ForegroundColor $([Constants]::MessageType.Error)
+                Write-Host "Skipping the Resource: [$($_.ResourceName)]..." -ForegroundColor $([Constants]::MessageType.Warning)
+                $logResource = @{}
+                $logResource.Add("ResourceGroupName",($_.ResourceGroupName))
+                $logResource.Add("ResourceName",($_.ResourceName))
+                $logResource.Add("Reason","Valid resource id(s) not found in input json file.")    
+                $logSkippedResources += $logResource
+                Write-Host $([Constants]::SingleDashLine)
+            }
+        }
+    }
+    else
+    {
+        # No file path provided as input to the script. Fetch all SQL Servers in the Subscription.
+        if ([String]::IsNullOrWhiteSpace($FilePath))
+        {
+            Write-Host "Fetching all SQL Servers in Subscription: $($context.Subscription.SubscriptionId)" -ForegroundColor $([Constants]::MessageType.Info)
+            Write-Host $([Constants]::SingleDashLine)
+            # Get all SQL Servers in the Subscription
+            $sqlServerResources = Get-AzSqlServer  -ErrorAction Stop
+
+       
+            $totalsqlServerResources = ($sqlServerResources | Measure-Object).Count
+        
+        }
+        else
+        {
+            if (-not (Test-Path -Path $FilePath))
+            {
+                Write-Host "ERROR: Input file - [$($FilePath)] not found. Exiting..." -ForegroundColor $([Constants]::MessageType.Error)
+                Write-Host $([Constants]::DoubleDashLine)
+                return
+            }
+
+            Write-Host "Fetching all SQL Servers(s) from [$($FilePath)]" -ForegroundColor $([Constants]::MessageType.Info)
```

Comment:
"..." at the end of statement as it shows that process is in progress.

Reply:
added